### PR TITLE
[FLINK-20722][hive] HiveTableSink should copy the record when convert…

### PR DESCRIPTION
**Pull request by lirui-apache**
…ing RowData to Row

## What is the purpose of the change

Fix the ClassCastException issue when inserting into hive.

## Brief change log

  - Copy the record when converting RowData to Row in HiveTableSink
  - Add test case

## Verifying this change

Existing and added test cases

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? NA

### CODE REVIEW

**Comment by lirui-apache** [comment]
Hi @godfreyhe , could you help review this PR? Thanks.

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/multipleinput/MultipleInputStreamOperatorBase.java [hunk 1]
```diff
@@ -245,9 +245,9 @@ private void createAllOperators(StreamOperatorParameters<RowData> parameters) {
                     output = outputs[0];
                 } else {
                     if (isObjectReuseEnabled) {
```
Comment (godfreyhe) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/multipleinput/MultipleInputStreamOperatorBase.java:247]: please add some comments for this behavior, just like the comments in OperatorChain#createOutputCollector method

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/TableEnvHiveConnectorITCase.java [hunk 2]
```diff
@@ -67,6 +67,43 @@ public static void setup() {
...
+                                    + "FROM (SELECT t1.key, p1.val\n"
+                                    + "      FROM src2 t1\n"
+                                    + "      LEFT OUTER JOIN src1 p1\n"
+                                    + "      ON (t1.key = p1.key)\n"
+                                    + "      UNION ALL\n"
+                                    + "      SELECT t2.key, p2.val\n"
+                                    + "      FROM src2 t2\n"
+                                    + "      LEFT OUTER JOIN src1 p2\n"
+                                    + "      ON (t2.key = p2.key)) j")
+                    .await();
```
Comment (godfreyhe) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/TableEnvHiveConnectorITCase.java:100]: do we need to check the result ?

**Review by godfreyhe — COMMENTED** [review]
Thanks for the contribution @lirui-apache , I left two minior comments

**Comment by lirui-apache** [comment]
@godfreyhe Thanks for your comments. PR updated accordingly.

**Review by godfreyhe — APPROVED** [review]
LGTM

**Pull request closed without merging**